### DISCUSSION TITLE
Adding an Factorio server

### DISCUSSION THREAD

**Discussion by MrPrecise** [AutoPacker-OSS/AutoPacker#73] [General]
Hey,

I was thinking about adding a Factorio server we talked about in [Issue 25](https://github.com/AutoPacker-OSS/autopacker/issues/25)

Earlier I was looking into docker images that we could use for the server and I found this that seems well established [Factorio Docker](https://github.com/factoriotools/factorio-docker), do you think we could use this?

After looking for this I got a bit stuck on what to do next, do you have any good place to start implementing this?

**Comment by ANicholasson** [comment]
We do have planned on creating a tag-based editor where you can define different "tags"/variables when adding new tools and technologies to be supported, but for now, it's more or less quite manual work.

### Process

1. The image you found looks quite good, and similar to the one we are using for Minecraft vanilla (which is a good start) so you can roll with that.
2. You will need to create an own docker-compose "block" for the factorio server. You can look at the one set up for Minecraft [here](file-delivery-api/data/templates/docker-compose/docker-compose-vanilla-minecraft.yml). So you would have to follow the instructions or try and make use of the information available at the docker image given in the link 
3. Related to the above: I would recommend you start with setting all the other environment variables by default and only enabling the port to be specified by the user.
4. You also need to add the "id" (hint: look at how the ids match the naming of the files) you give it to this list: https://github.com/AutoPacker-OSS/autopacker/blob/9197f373d915089f3086558b067d2ae5f674ecae/file-delivery-api/src/main/java/no/autopacker/filedeliveryapi/init/DBImportDefaultDockerFilesCompose.java#L37 (this is only temporary)
5. Then you need to implement the frontend part. This is manually done atm so you should look at how it has been done with Minecraft [here](https://github.com/AutoPacker-OSS/autopacker/blob/develop/web-application/src/views/ProfileDashboard/AddModule/GameModule/components/Setup/Setup.js)

This might look like a lot, but there shouldn't take to long once you get the grasp of how things work together.

As stated initially, we _hope_ we are able to switch this out with a "tag-based upload editor" for a user with a specific privilege later on. This subproject might even be of interest?
